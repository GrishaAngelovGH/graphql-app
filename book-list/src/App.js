import Header from "components/Header"

function App() {
  return (
    <div className="row g-0 vh-100 bg-body-secondary">
      <div className="col-md-12">
        <Header />
      </div>
    </div>
  );
}

export default App;
