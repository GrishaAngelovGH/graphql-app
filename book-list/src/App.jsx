import Header from "components/Header"
import BookList from "components/BookList"

function App() {
  return (
    <div className="row g-0 vh-100 bg-body-secondary">
      <div className="col-md-12">
        <Header />
        <BookList />
      </div>
    </div>
  );
}

export default App;
