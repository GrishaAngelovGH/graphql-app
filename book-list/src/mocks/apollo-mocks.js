import { gql } from '@apollo/client'

const mocks = [
  {
    request: {
      query: gql`
      query GetBooks {
        books {
          isbn
          title
        }
      }
    `
    },
    result: {
      data: {
        books: [
          { isbn: '1', title: 'Title 1' },
          { isbn: '2', title: 'Title 2' },
          { isbn: '3', title: 'Title 3' }
        ]
      }
    }
  },
  {
    request: {
      query: gql`
      query GetAuthors {
        authors {
          id
          name
        }
      }
    `
    },
    result: {
      data: {
        authors: [
          { id: '1', name: 'Author 1' },
          { id: '2', name: 'Author 2' }
        ]
      }
    }
  },
  {
    request: {
      query: gql`
      query Book($isbn: String) {
        book(isbn: $isbn) {
          title
          author {
            name
            books{
              isbn
              title
            }
          }
        }
      }
    `
    },
    result: {
      data: {
        book: {
          title: 'Title 1',
          author: {
            name: 'Author 1',
            books: [
              { isbn: 1, title: 'Title 1' },
              { isbn: 2, title: 'Title 2' }
            ]
          }
        }
      }
    }
  }
]

export default mocks