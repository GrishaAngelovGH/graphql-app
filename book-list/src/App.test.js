import { MockedProvider } from '@apollo/client/testing'
import { act, render } from '@testing-library/react'
import mocks from 'mocks/apollo-mocks'

import App from './App'

test('should render App component', async () => {
  const view = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <App />
    </MockedProvider>
  )

  await act(() => new Promise(resolve => setTimeout(resolve, 0)))

  expect(view).toMatchSnapshot()
})