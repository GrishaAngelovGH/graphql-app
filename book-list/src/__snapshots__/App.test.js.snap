// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row g-0 vh-100 bg-body-secondary"
      >
        <div
          class="col-md-12"
        >
          <header>
            <div
              class="overlay"
            >
              <h1>
                Book List
              </h1>
            </div>
          </header>
          <div
            class="row g-0"
          >
            <div
              class="col-md-7 overflow-auto"
              style="height: calc(100vh - 200px);"
            >
              <h3
                class="text-center text-secondary fw-bold"
              >
                Books
              </h3>
              <button
                class="mx-2 mt-2 fs-6 btn btn-primary"
                type="button"
              >
                Title 1
              </button>
              <button
                class="mx-2 mt-2 fs-6 btn btn-primary"
                type="button"
              >
                Title 2
              </button>
              <button
                class="mx-2 mt-2 fs-6 btn btn-primary"
                type="button"
              >
                Title 3
              </button>
              <div
                class="toast-container top-0 end-0 p-3"
              />
              <button
                class="fixed-bottom w-25 btn btn-outline-success"
                style="left: 20px; bottom: 20px;"
                type="button"
              >
                Add New Book
              </button>
            </div>
            <div
              class="col-md-5"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row g-0 vh-100 bg-body-secondary"
    >
      <div
        class="col-md-12"
      >
        <header>
          <div
            class="overlay"
          >
            <h1>
              Book List
            </h1>
          </div>
        </header>
        <div
          class="row g-0"
        >
          <div
            class="col-md-7 overflow-auto"
            style="height: calc(100vh - 200px);"
          >
            <h3
              class="text-center text-secondary fw-bold"
            >
              Books
            </h3>
            <button
              class="mx-2 mt-2 fs-6 btn btn-primary"
              type="button"
            >
              Title 1
            </button>
            <button
              class="mx-2 mt-2 fs-6 btn btn-primary"
              type="button"
            >
              Title 2
            </button>
            <button
              class="mx-2 mt-2 fs-6 btn btn-primary"
              type="button"
            >
              Title 3
            </button>
            <div
              class="toast-container top-0 end-0 p-3"
            />
            <button
              class="fixed-bottom w-25 btn btn-outline-success"
              style="left: 20px; bottom: 20px;"
              type="button"
            >
              Add New Book
            </button>
          </div>
          <div
            class="col-md-5"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
